docs/en/sql-reference/functions/array-functions.md

```diff
@@ -1400,21 +1629,91 @@ Result:
 └────────────────────────────────┘
 ```
 
-## arrayEnumerateDense(arr)
+## arrayEnumerateDense
 
 Returns an array of the same size as the source array, indicating where each element first appears in the source array.
 
-Example:
+**Syntax**
+
+```sql
+arrayEnumerateDense(arr)
+```
+
+**Example**
+
+Query:
 
 ``` sql
 SELECT arrayEnumerateDense([10, 20, 10, 30])
 ```
 
+Result:
+
 ``` text
 ┌─arrayEnumerateDense([10, 20, 10, 30])─┐
 │ [1,2,1,3]                             │
 └───────────────────────────────────────┘
 ```
+## arrayEnumerateDenseRanked
```

Comment:
Can you add a comment here for what use case this function would be useful for?

Reply:
I think this one together with `arrayEnumerateUniqRanked` probably wouldn't be used by the average user - there is a comment in the code "This is very unusual function made as a special order for our dear customer - Metrica web analytics system", not entirely sure what application they had in mind.

Reply:
Yes, this is a special purpose function but I disagree that we need to document a use case. An example (as included below) is just fine.